projectile-rails.el

```diff
@@ -35,7 +35,7 @@
 ;;
 ;;; Code:
 
-(require 'cl)
+(eval-when-compile (require 'cl-lib))
```

Comment:
@wkirschbaum please enlighten me, why the use of `eval-when-compile`?

Reply:
@asok As far as I understand eval-when-compile will work like progn when this file is interpreted, but when byte compiled it won't expose functions from cl-lib during runtime and just expand the macros in this file, which means that this package won't require cl-lib functions for everyone else. I might understand it incorrectly or it might be unnecessary, so happy to remove it. I see some packages do it and some don't so :shrug: I am new to elisp, so any insight on this will also be appreciated. 

I also need to mention that this package won't be compatible for < 24.3, so not sure what the policy is around backward compatibility.